pulsar-broker-common/src/main/java/com/yahoo/pulsar/broker/authentication/AuthenticationProviderAthenz.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.pulsar.broker.authentication;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.util.List;
+import java.security.PublicKey;
+
+import javax.naming.AuthenticationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+import com.yahoo.athenz.auth.token.RoleToken;
+import com.yahoo.athenz.zpe.AuthZpeClient;
+import com.yahoo.pulsar.broker.ServiceConfiguration;
+import com.yahoo.pulsar.broker.authentication.AuthenticationDataSource;
+import com.yahoo.pulsar.broker.authentication.AuthenticationProvider;
+
+public class AuthenticationProviderAthenz implements AuthenticationProvider {
+
+    private static final String DOMAIN_NAME_LIST = "athenzDomainNames";
+
+    private List<String> domainNameList = null;
+
+    @Override
+    public void initialize(ServiceConfiguration config) throws IOException {
+        if (config.getProperty(DOMAIN_NAME_LIST) == null) {
+            throw new IOException("No athenz domain name specified");
+        }
+        String domainNames = (String) config.getProperty(DOMAIN_NAME_LIST);
+        domainNameList = Lists.newArrayList(domainNames.split(","));
+        log.info("Supported domain names for athenz: {}", domainNameList);
+    }
+
+    @Override
+    public String getAuthMethodName() {
+        return "athenz";
+    }
+
+    @Override
+    public String authenticate(AuthenticationDataSource authData) throws AuthenticationException {
```

Comment:
How does this auth provider interacts with the authorization policies in Pulsar? Do one needs to "grant permission" on the namespace in Pulsar or everything is handled in Athenz?